apis/externalsecrets/v1beta1/secretstore_pulumi_types.go

```diff
@@ -0,0 +1,42 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	esmeta "github.com/external-secrets/external-secrets/apis/meta/v1"
+)
+
+type PulumiProvider struct {
+	// APIURL is the URL of the Pulumi API.
+	APIURL string `json:"apiUrl,omitempty"`
+
+	// Auth configures how secret-manager authenticates with a Pulumi instance.
+	AccessToken *PulumiProviderSecretRef `json:"accessToken"`
+
+	Organization string `json:"organization"`
+	Environment  string `json:"environment"`
+
+	// Duration is the duration for which the session is valid in minutes.
+	// Defaults to 5 minutes.
+	// +kubebuilder:default="5m"
+	Duration *metav1.Duration `json:"duration,omitempty"`
```

Comment:
Do we really want/need to expose this on the CRD level? 

The API docs are not very precise on the implications of this value. Can we open an >environment hundreds of times within a `30s` time window?

I guess i would prefer not to expose it on the CRD level unless we have to. That would simplify the code a bit as well.

Reply:
Removed, thats fine for us!